### PR TITLE
Fix bug - Manually entered split transactions are not cleared on import #200

### PR DESCRIPTION
**Pull request by kstockk**
Relates to https://github.com/actualbudget/actual/issues/200

### CODE REVIEW

**Comment by kyrias** [comment]
- Link an account to the GoCardless DEMO bank.
- Select one of the imported transactions and create one with the same duplicates, but make it a split transaction.
- Delete the original.
- Sync again.  Transaction is marked as cleared but the cleared total does not reflect this transaction.

File: packages/loot-core/src/server/accounts/sync.ts [hunk 1]
```diff
@@ -537,6 +537,20 @@ export async function reconcileGoCardlessTransactions(acctId, transactions) {
...
+    LEFT JOIN v_transactions p ON t.parent_id = p.id
+    WHERE t.is_child = 1 AND t.cleared != p.cleared
+  `);
+
+  let updatedClearedRows = clearedRows.map(row => ({
+    id: row.id,
+    cleared: row.cleared === 1,
+  }));
+  await batchUpdateTransactions({ updated: updatedClearedRows });
+
```
Comment (kyrias) [inline packages/loot-core/src/server/accounts/sync.ts:553]: Having to query every transaction like this seems very inefficient.  I think the following would be a better solution:

```diff
diff --git a/packages/loot-core/src/server/accounts/sync.ts b/packages/loot-core/src/server/accounts/sync.ts
index 4733dc6f..4cbea6d5 100644
--- a/packages/loot-core/src/server/accounts/sync.ts
+++ b/packages/loot-core/src/server/accounts/sync.ts
@@ -431,7 +431,7 @@ export async function reconcileGoCardlessTransactions(acctId, transactions) {
       // matched transaction. See the final pass below for the needed
       // fields.
       fuzzyDataset = await db.all(
-        `SELECT id, date, imported_id, payee, category, notes FROM v_transactions
+        `SELECT id, is_parent, date, imported_id, payee, category, notes FROM v_transactions
            WHERE date >= ? AND date <= ? AND amount = ? AND account = ? AND is_child = 0`,
         [
           db.toDateRepr(monthUtils.subDays(trans.date, 4)),
@@ -509,6 +509,17 @@ export async function reconcileGoCardlessTransactions(acctId, transactions) {
       if (hasFieldsChanged(existing, updates, Object.keys(updates))) {
         updated.push({ id: existing.id, ...updates });
       }
+
+      if (existing.is_parent && existing.cleared !== updates.cleared) {
+        const children = await db.all(
+          'SELECT * FROM v_transactions WHERE parent_id = ?',
+          [existing.id],
+        );
+        for (const child of children) {
+          updated.push({ id: child.id, cleared: updates.cleared });
+        }
+      }
     } else {
       // Insert a new transaction
       let finalTransaction = {
```

Comment (kstockk) [reply packages/loot-core/src/server/accounts/sync.ts]: I agree. Thanks for for the suggestion

I've updated it in both the reconcileGoCardlessTransactions and reconcileTransactions functions

Please review

**Comment by kstockk** [comment]
To test without GoCardless.

Export Account's csv.
Import the csv back into the account -> This should mark all transactions as cleared
Undo to mark them as uncleared
Edit some transactions by making them split transactions
Import the csv back into the account -> this should mark all transactions as cleared (including the split transactions)

**Review by shall0pass — APPROVED** [review]
Looks good to me!

**Pull request merged**